docs/reference/target-group-policy.md

```diff
@@ -0,0 +1,94 @@
+# TargetGroupPolicy API Reference
+
+## TargetGroupPolicy
+
+By default, AWS Gateway API Controller assumes plaintext HTTP/1 traffic for backend Kubernetes resources.
+TargetGroupPolicy is a CRD that can be attached to a Service, which allows the users to define protocol and
+health check configurations of those backend resources.
+
+When attaching a policy to a resource, the following restrictions apply:
+
+* A policy can be only attached to `Service` resources.
+* The attached resource can only be `backendRef` of `HTTPRoute` and `GRPCRoute`.
+* The attached resource should exist in the same namespace as the policy resource.
+
+The policy will not take effect if:
+
+* The resource does not exist
+* The resource is not referenced by any route
+* The resource is referenced by a route of unsupported type
+
+Attaching TargetGroupPolicy to a resource that is already referenced by a route may result in a replacement
+of VPC Lattice TargetGroup resource.
+
+|Field	|Description	|
+|---	|---	|
+|`apiVersion` *string*	|``application-networking.k8s.aws/v1alpha1``	|
+|`kind` *string*	|``TargetGroupPolicy``	|
+|`metadata` [*ObjectMeta*](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#objectmeta-v1-meta)	|Kubernetes metadata for the resource.	|
+|`spec` *TargetGroupPolicySpec*	|Defines the desired state of TargetGroupPolicy.	|
+
+## TargetGroupPolicySpec
+
+Appears on: TargetGroupPolicy
+
+TargetGroupPolicySpec defines the desired state of TargetGroupPolicy.
+
+Updates to certain fields of this configuration result in a replacement of VPC Lattice TargetGroup resource.
```

Comment:
I would avoid vague mentions like "certain fields. It's better to list fields that can cause replacement here, even it's duplicated in the table below, saying something important twice wont hurt.